src/cocotb_tools/pytest/runner.py

```diff
@@ -0,0 +1,64 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Module related with handling cocotb runners and collecting cocotb tests from these runners."""
+
+from __future__ import annotations
+
+import os
+from collections.abc import Iterable, Sequence
+from importlib import import_module
+from pathlib import Path
+
+from pytest import Collector, Item, Module
+
+
+class Runner(Collector):
+    """Collector that will collect cocotb tests from cocotb runner."""
+
+    def __init__(
+        self,
+        item: Item,
+        *args,
+        **kwargs,
+    ):
```

Comment:
Add type annotations.

Reply:
@ktbarrett I have added that in another commit `chore(name): mypy`. I was forced by `mypy` when I have added `cocotb_tools.pytest` in the `pyproject.toml` file in this commit ae466e07ee0076d3bc7fdcb3baa707ece4e34507